daemon-scheduler/pkg/deployment/deployment.go

```diff
@@ -151,10 +151,11 @@ func (d deployment) CreateSubDeployment(ctx context.Context, environmentName str
 			"There is no deployment for environment with name '%s' to create a sub-deployment", environmentName)
 	}
 
-	return d.startSubDeployment(ctx, env, deployment, instanceARNs)
+	return d.startDeployment(ctx, env, deployment, instanceARNs)
```

Comment:
I see above that deployment = d.GetCurrentDeployment(). Does it now pick the pending deployment first? And if that isn't present, looks for the completed deployment?